promql/engine.go

```diff
@@ -129,6 +129,8 @@ type Query interface {
 	Statement() parser.Statement
 	// Stats returns statistics about the lifetime of the query.
 	Stats() *stats.QueryTimers
+	// SampleStats returns sample statistics about the lifetime of the query.
+	SampleStats() *stats.QuerySamples
```

Comment:
We should probably extend the above function rather than adding a new one.

Reply:
Done. Extended the `Stats()`